pkg/api/v1/types.go

```diff
@@ -1045,9 +1045,21 @@ func newPublicAnexiaCloudSpec(internal *kubermaticv1.AnexiaCloudSpec) (public *P
 type ClusterStatus struct {
 	// Version actual version of the kubernetes master components
 	Version ksemver.Semver `json:"version"`
-
 	// URL specifies the address at which the cluster is available
 	URL string `json:"url"`
+	// CCM represents the externalCCM status
+	CCM ExternalCCMStatus `json:"ccm"`
+}
+
+// ExternalCCMStatus represents the external CCM migration status
+// swagger:model ExternalCCMStatus
+type ExternalCCMStatus struct {
+	// the ClusterFeatureExternalCloudProvider in the cluster .spec.features
+	ExternalCCM bool `json:"externalCCM"`
+	// the logic AND of the CCMMigrationNeededAnnotation and CSIMigrationNeededAnnotation annotations
+	MigrationNeeded *bool `json:"migrationNeeded,omitempty"`
+	// reflect the ClusterConditionCSIKubeletMigrationCompleted condition of the cluster
+	MigrationCompleted *bool `json:"migrationCompleted,omitempty"`
```

Comment:
Tiny nit: I think those should be proper Go comments (e.g. `ExternalCCM represents...`).